src/main/io/serial_command_interpreter.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of Cleanflight, Betaflight and INAV
+ *
+ * Cleanflight, Betaflight and INAV are free software. You can
+ * redistribute this software and/or modify this software under
+ * the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * Cleanflight, Betaflight and INAV are distributed in the hope that
+ * they will be useful, but WITHOUT ANY WARRANTY; without even the
+ * implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "platform.h"
+
+#include "io/serial_command_interpreter.h"
+
+void serialCommandInterpreterInit(serialCommandInterpreter_t *interpreter)
+{
+    interpreter->lastPlus = 0;
+    interpreter->state = SERIAL_COMMAND_INTERPRETER_STATE_IDLE;
+}
+
+serialCommandInterpreterResult_e serialCommandInterpreterUpdate(serialCommandInterpreter_t *interpreter, uint8_t c)
+{
+    serialCommandInterpreterResult_e result = SERIAL_COMMAND_INTERPRETER_RESULT_NONE;
+    switch (interpreter->state) {
+    case SERIAL_COMMAND_INTERPRETER_STATE_IDLE:
+        if (c == '+') {
+            interpreter->lastPlus = millis();
+            interpreter->state = SERIAL_COMMAND_INTERPRETER_STATE_PLUS_ONE;
+        }
+        break;
+    case SERIAL_COMMAND_INTERPRETER_STATE_PLUS_ONE:
+    case SERIAL_COMMAND_INTERPRETER_STATE_PLUS_TWO:
+        if (c == '+') {
+            timeMs_t now = millis();
+            timeMs_t delta = now - interpreter->lastPlus;
+            if (delta >= 900 && delta <= 1100) {
```

Comment:
It is impossible/hard for human to meet this timing

Reply:
Just realised I got the timing wrong. The original hayes is (pause > 1s)+++(pause > 1s). Then you're in command mode. That's still difficult to trigger by chance when transferring arbitrary data, but it's easy to trigger by the user. I'll update the PR.